### PR TITLE
[#112641071] Convert AzureRM to a 'shell' module with version range dependencies

### PR DESCRIPTION
**Pull request by hovsepm**

### CODE REVIEW

File: tools/AzureRM/AzureRM.psm1 [hunk 1]
```diff
@@ -1,260 +0,0 @@
...
-        Write-Output "$moduleName imported..." 
-      } catch {
-        Write-Warning "Skipping $Name module..."
-        Write-Warning $_
-      }
-    }
-  }
-}
-
-function Uninstall-ModuleWithVersionCheck([string]$Name,[string]$MinVersion)
```
Comment (markcowl) [inline tools/AzureRM/AzureRM.psm1]: We *may* decide that we need an uninstall cmdlet - currently, if you uninstall the manifest module, the dependent modules remain.  We may retain an 'Uninstall-AzureRm' script which uninstalls the module and its dependencies (not sure if it's possible for a cmdlet to remove all associated modules.  Regardless, we should decide on some acceptance criteria before merging.

File: tools/AzureRM/AzureRM.psd1 [hunk 2]
```diff
@@ -48,7 +45,36 @@ CLRVersion='4.0'
 ProcessorArchitecture = 'None'  
   
 # Modules that must be imported into the global environment prior to importing this module  
-RequiredModules = @() 
+RequiredModules = @(
+    @{ ModuleName = 'AzureRM.Profile'; MaximumVersion = '1.9999.9999.9999'; ModuleVersion = '1.0.4'},
```
Comment (markcowl) [inline tools/AzureRM/AzureRM.psd1]: Is there any script to update the Module Version dependencies in this module (was there any such script for updating the versions in the script)?

**Comment by hovsepm** [comment]
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/263/

**Comment by hovsepm** [comment]
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/266/

**Comment by hovsepm** [comment]
sign: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/271/

**Comment by hovsepm** [comment]
sign: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/272/

**Comment by hovsepm** [comment]
sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/293/

**Comment by hovsepm** [comment]
singing the sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/294/

**Comment by hovsepm** [comment]
singing the sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/295/

**Comment by hovsepm** [comment]
latest sign job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/299/

**Comment by hovsepm** [comment]
sign in super job: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/300/

**Comment by hovsepm** [comment]
The sign job! http://azuresdkci.cloudapp.net/job/powershell-sign/306/

**Comment by hovsepm** [comment]
Sin http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-sign/308/

**Comment by hovsepm** [comment]
the job:

**Comment by hovsepm** [comment]
demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/587/

**Comment by hovsepm** [comment]
fixed demand: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-demand/588/

**Pull request merged**